mlflow/transformers.py

```diff
@@ -1675,8 +1674,7 @@ def trim_input(data_in, data_out):
                 # If the user has indicated to remove newlines and extra spaces from the generated
                 # text, replace them with a single space.
                 if collapse_whitespace:
-                    for to_replace, replace in replacements.items():
-                        data_out = re.sub(to_replace, replace, data_out).strip()
+                    data_out = re.sub(r"\s+", " ", data_out).strip()
```

Comment:
@BenWilson2 We collapse whitespace characters if `data_out.startswith(data_in + "\n\n")`. Is this intended?

Reply:
The original intention with nesting that was to only apply the whitespace collapsing to the InstructionTextGenerationPipeline (Dolly), but I think it's valuable for any of the TextGeneration types.

Reply:
Got it. I'll merge this PR. We can make that change in a separate PR

Reply:
running full test suite locally to validate this logic change

Reply:
I'll file the follow-on change in a few minutes after merge.